platform-hub-api/app/services/kubernetes/static_token_service.rb

```diff
@@ -0,0 +1,82 @@
+module Kubernetes
+  module StaticTokenService
+    extend self
+
+    # Static tokens are stored in HashRecord as:
+    # data: [
+    #   {"token" => "token1", user" => "user1", "uid" => "uid1", "groups" => ["group1","group2"]}
+    #   {"token" => "token2", user" => "user2", "uid" => "uid2", "groups" => []}
+    #   ...
+    # ]
+    # Note: All tokens are encrypted!
+
+    def create_or_update(cluster, kind, user_name, groups = [])
+      static_tokens = get_static_tokens(cluster, kind)
+
+      record = static_tokens.data.find do |t|
+        t['user'] == user_name.to_s
+      end
+
+      if record.nil?
+        token = Kubernetes::TokenService.generate_secure_random
+        record = Hashie::Mash.new(
+          token: ENCRYPTOR.encrypt(token),
+          user: user_name,
+          uid: Kubernetes::TokenService.generate_secure_random,
+          groups: Kubernetes::TokenService.send(:cleanup, groups)
+        )
+
+        static_tokens.data << record
+      else
+        record['groups'] = groups.present? ? 
+          Kubernetes::TokenService.send(:cleanup, groups) : []
+      end
+
+      static_tokens.save!
+      "Created/updated #{kind} account for `#{user_name}` (token: #{ENCRYPTOR.decrypt(record['token'])})"
```

Comment:
One thing to consider with messages like this: if they are getting logged out to STDOUT then they'll show up in ELK, and thus we'll be leaking sensitive info. 

**The same issue applies for audits – for better or worse, [I've made it so that all Audit entry creations get logged out to STDOUT](https://github.com/UKHomeOffice/platform-hub/blob/master/platform-hub-api/app/models/audit.rb#L10) – we should probably rethink this?** (So far I've tried not to put anything too sensitive in those fields that do get logged out [e.g. `comment`], so maybe just need to check that we're not outputting anything sensitive in [the fields being logged to STDOUT](https://github.com/UKHomeOffice/platform-hub/blob/master/platform-hub-api/app/models/audit.rb#L21-L31))